src/pages/[spaceId]/index.tsx

```diff
@@ -53,31 +83,18 @@ export const getStaticProps = getCommonStaticProps<
     const queryClient = new QueryClient()
 
     let { spaceId: paramSpaceId } = context.params ?? {}
-    const spaceIdOrTopic = (paramSpaceId as string) ?? getMainSpaceId()
-    let spaceId = spaceIdOrTopic
-    if (isNaN(parseInt(spaceIdOrTopic))) {
-      const spaceIdFromTopic = getSpaceIdFromTopic(spaceIdOrTopic)
-      if (spaceIdFromTopic) {
-        spaceId = spaceIdFromTopic
-      } else {
-        return undefined
-      }
-    }
+    const spaceId = getSpaceIdFromParam(paramSpaceId as string)
+    if (!spaceId) return undefined
 
     try {
       const subsocialApi = await getSubsocialApi()
       const postIds = await subsocialApi.blockchain.postIdsBySpaceId(spaceId)
       const allPostIds = [...postIds, ...getLinkedPostIdsForSpaceId(spaceId)]
 
-      const promises = allPostIds.map((postId) => {
-        return subsocialApi.blockchain.getReplyIdsByPostId(postId)
-      })
-      const postsPromise = getPostsFromCache(allPostIds)
-
-      const commentIdsByPostId = await Promise.all(promises)
-      const posts = await postsPromise
-
-      const lastPosts = await getLastPosts(commentIdsByPostId)
+      const [{ lastPosts, posts, commentIdsByPostId }] = await Promise.all([
+        getPostsData(allPostIds),
+        prefetchBlockedEntities(queryClient, allPostIds),
+      ] as const)
```

Comment:
Same code exists in [...topic] page

Reply:
hmm yes, but the getPostsData is different, so I think its not really extractable